### PR TITLE
PyPlot- and PlanarRigidBodyVisualizer

### PR DESCRIPTION
**Pull request by gizatt**
Adds:
- Pendulum, double_pendulum URDFs, and double_pendulum SDF (the latter doesn't work very well, as it has no transmissions and so no control inputs...) for testing
- `pyplot_visualizer.py`, a base class for visualization that doesn't do much other than maintain a pyplot figure with a periodic publish. `pendulum_visualizer.py` subclasses this and so has been simplified a tiny bit
- 'planar_rigid_body-visualizer.py', which takes a RigidBodyTree, and uses its state as input to draw a planar visualization via a specified projection matrix.

*Relies on Drake PR #7913*

### CODE REVIEW

File: src/double_pendulum.sdf [hunk 1]
```diff
@@ -0,0 +1,181 @@
+<?xml version='1.0' ?>
```
Comment (RussTedrake) [inline src/double_pendulum.sdf]: why add the sdf, too?

File: src/planar_rigid_body_visualizer.py [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+                self.body_fill_list[body_fill_index].get_path().vertices[:, :] = np.transpose(patch)
+                body_fill_index += 1
+
+if __name__ == "__main__":
+    # Usage demo: load a URDF, rig it up with a constant torque input, and draw it.
+
+    np.set_printoptions(precision=5, suppress=True)
+
+    '''
+    rbt = RigidBodyTree("Pendulum.urdf", floating_base_type=pydrake.rbtree.FloatingBaseType.kFixed)
```
Comment (RussTedrake) [inline src/planar_rigid_body_visualizer.py]: Pendulum.urdf => pendulum.urdf

File: src/planar_rigid_body_visualizer.py [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+        body_fill_index = 0
+        for body_i in range(self.rbtree.get_num_bodies()-1):
+            tf = self.rbtree.relativeTransform(kinsol, 0, body_i+1)
+            viewPatches, _ = self.getViewPatches(body_i, tf)
+            for patch in viewPatches:
+                self.body_fill_list[body_fill_index].get_path().vertices[:, :] = np.transpose(patch)
+                body_fill_index += 1
+
+if __name__ == "__main__":
+    # Usage demo: load a URDF, rig it up with a constant torque input, and draw it.
```
Comment (RussTedrake) [inline src/planar_rigid_body_visualizer.py]: add it to the CMakeLists.txt as a unit test?  (ideally with small sim durations, so we know if it breaks, but don't chew up frivolous cycles on circleci)

File: src/planar_rigid_body_visualizer.py [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+    Tview = np.array([[1., 0., 0., 0.], [0., 0., 1., 0.], [0., 0., 0., 1.]], dtype=np.float64)
+    pbrv = PlanarRigidBodyVisualizer(rbt, Tview, [-1.2, 1.2], [-1.2, 1.2])
+    '''
+
+    rbt = RigidBodyTree("double_pendulum.urdf", floating_base_type=pydrake.rbtree.FloatingBaseType.kFixed)
+    Tview = np.array([[1., 0., 0., 0.], [0., 0., 1., 0.], [0., 0., 0., 1.]], dtype=np.float64)
+    pbrv = PlanarRigidBodyVisualizer(rbt, Tview, [-3.0, 3.0], [-3.0, 3.0], use_random_colors=True)
+
+    '''
+    # Doesn't work correctly, as it has no inputs hooked up.
```
Comment (RussTedrake) [inline src/planar_rigid_body_visualizer.py]: you mean the sdf parser doesn't parse actuators?  (and there aren't actuators in the sdf).

Comment (gizatt) [reply src/planar_rigid_body_visualizer.py]: `sdf_parser` is happy to populate actuators (or so it looks from the source), but the double pendulum SDF from `drake/examples/double_pendulum` has effort limits of 0 on its two revolute joints, which tells the SDF parser to not assign actuators to those joints.

File: src/double_pendulum.sdf (hunk 1 above)
Comment (gizatt) [reply src/double_pendulum.sdf]: Original intent was to establish that I could load from SDFs too, but I agree that there's little point having both here, and the Val loading example looks similar enough to loading an SDF anyway in terms of setting up frames.

File: src/planar_rigid_body_visualizer.py (hunk 2 above)
Comment (gizatt) [reply src/planar_rigid_body_visualizer.py]: Done!

**Comment by RussTedrake** [comment]
@gizatt -- I think that the prereqs have landed in drake nightlies.  
it looks like i also added matplotlib to the install scripts, and also added a method to src/CMakeLists.txt to make the python tests w/ matplotlib "silent" (they won't actually render a window, so should run faster).  can you update and we'll get this in?

**Comment by gizatt** [comment]
Got it, those tweaks should be in (pending CI)

Still in here is adding matplotlib to the Mac prereqs list -- I had to guess how to do it, though, as I don't have a mac to test on.

**Comment by RussTedrake** [comment]
i added matplotlib to mac prereqs already.  (via pip instead of homebrew)

File: scripts/setup/mac/install_prereqs [hunk 5]
```diff
@@ -71,6 +71,7 @@ ipopt
 libccd
 libyaml
 nlopt
+matplotlib
```
Comment (RussTedrake) [inline scripts/setup/mac/install_prereqs]: please zap

**Comment by gizatt** [comment]
Zapped

**Pull request merged**